src/server/stream_family.cc

```diff
@@ -1407,25 +1581,45 @@ std::optional<ReadOpts> ParseReadArgsOrReply(CmdArgList args, ConnectionContext*
     string_view key = ArgS(args, i - streams_count);
     string_view idstr = ArgS(args, i);
 
+    StreamIDsItem sitem;
     ParsedStreamId id;
 
     if (idstr == "$") {
       // Set ID to 0 so if the ID cannot be resolved (when the stream doesn't
       // exist) it takes the first entry added.
+      if (opts.read_group) {
+        (*cntx)->SendError("The $ can be specified only when calling XREAD.", kSyntaxErr);
+        return std::nullopt;
+      }
       id.val.ms = 0;
       id.val.seq = 0;
       id.last_id = true;
-      opts.stream_ids.emplace(key, id);
+      sitem.id = id;
+      auto [_, is_inserted] = opts.stream_ids.emplace(key, sitem);
+      if (!is_inserted) {
+        (*cntx)->SendError("Same stream specified multiple time");
```

Comment:
`"Same stream specified multiple time"` appears multiple times (the irony).
Please factor out into a constant and use it everywhere.